tools/ts-generator/types/types_gen_js.py

```diff
@@ -60,7 +60,7 @@
 {%- endmacro -%}
 
 {%- macro write_varint(field, propertyPath, buffer, assign) %}
-    {{ "writtenBytes = " if assign -}}
+    {{ "writtenBytes += " if assign -}}
```

Comment:
why is this change in the PR?

Reply:
This is a fix for writeVarint, this change is necessary for run `demo.ts` as expected.
context: the first and second commit on this PR are fixing for the `demo.ts` file because the last commit needs run the `demo` script in order to check the Optional type.